libs/sarif/src/schema/message.rs

```diff
@@ -0,0 +1,482 @@
+/// removing the line breaks so that the entire message renders on a single line.
+///
+/// The message string **may** contain [placeholders] ([§3.11.5]) and [embedded links] ([§3.11.6]).
+///
+/// If the message consists of more than one sentence, its first sentence **should** provide a
+/// useful summary of the message, suitable for display in cases where UI space is limited.
+///
+/// > ## Note 1
+/// >
+/// > If a tool does not construct the message in this way, the initial portion of the message that
+/// > a viewer displays where UI space is limited might not be understandable.
+///
+/// > ## Note 2
+/// >
+/// > The rationale for these guidelines is that the SARIF format is intended to make it feasible to
+/// > merge the outputs of multiple tools into a single user experience. A uniform approach to
+/// > message authoring enhances the quality of that experience.
+///
+/// A SARIF post-processor **should not** modify line break sequences (except perhaps to adapt them
+/// to a particular viewing environment).
+///
+///
+/// # Formatted messages
+/// [formatted messages]: #formatted-messages
+///
+/// ## General
+///
+/// Formatted messages **may** be of arbitrary length and **may** contain formatting information.
+/// The message string **may** also contain [placeholders] ([§3.11.5]) and [embedded links]
+/// ([§3.11.6]).
+///
+/// Formatted messages **shall** be expressed in [GitHub-Flavored Markdown][GFM]. Since GFM is a
+/// superset of [CommonMark][CMARK], any CommonMark Markdown syntax is acceptable.
+///
+///
+/// ## Security implications
+///
+/// For security reasons, SARIF producers and consumers **shall** adhere to the following:
+///
+/// - SARIF producers **shall not** emit messages that contain HTML, even though all variants of
+///   Markdown permit it.
+/// - Deeply nested markup can cause a stack overflow in the [Markdown processor][GFMENG]. To reduce
+///   this risk, SARIF consumers **shall** use a Markdown processor that is hardened against such
+///   attacks.
+///   > ## Note
+///   >
+///   > One example is the GitHub fork of the [cmark Markdown processor][GFMCMARK].
+/// - To reduce the risk posed by possibly malicious SARIF files that do contain arbitrary HTML
+///   (including, for example, `javascript:` links), SARIF consumers **shall** either disable HTML
+///   processing (for example, by using an option such as the `--safe` option in the cmark Markdown
+///   processor) or run the resulting HTML through an HTML sanitizer.
+///
+/// SARIF consumers that are not prepared to deal with the security implications of formatted
+/// messages **shall not** attempt to render them and **shall** instead fall back to the
+/// corresponding plain text messages.
+///
+///
+/// # Messages with placeholders
+/// [placeholders]: #messages-with-placeholders
+///
+/// A message string **may** include one or more “placeholders". The syntax of a placeholder is:
+///
+/// ```text
```

Comment:
```suggestion
/// ```ebnf
```

Reply:
SARIF uses ISO EBNF (https://www.iso.org/standard/26153.html) in contrast to RFC ABNF (https://datatracker.ietf.org/doc/html/rfc5234)